### PR TITLE
MD364 - Adds session key management proxy type

### PR DESCRIPTION
**Pull request by augustocollado**
This PR allows the delegating of key management
It adds:

- New proxy type 'SessionKeyManagement' in Dancebox runtime
- It adds unit testing for this new type

### CODE REVIEW

**Comment by girazoki** [comment]
@augustocollado you should not need any cargo.lock changes to make this PR, make sure there are not please, I think your tests are not passing because of these

**Review by girazoki — APPROVED** [review]
LGTM

File: runtime/dancebox/src/lib.rs [hunk 1]
```diff
@@ -1057,6 +1057,8 @@ pub enum ProxyType {
     Registrar = 6,
     /// Allow extrinsics related to Registrar that needs to be called through Sudo
     SudoRegistrar = 7,
+    /// Allow extrinsics from the Session pallet for key management.
+    SessionKeyManagement = 8,
```
Comment (tmpolaczyk) [inline runtime/dancebox/src/lib.rs:1061]: We don't need this in flashbox?

Comment (augustocollado) [reply runtime/dancebox/src/lib.rs:1061]: I don't think so. 
This came up as a requirement for node operators to decouple staking from key management, and there will be no external node operators on flashbox.

Comment (girazoki) [reply runtime/dancebox/src/lib.rs:1061]: Yeah exactly we could put it but no-one is going to use it

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**